src/backend/DataManager.ts

```diff
@@ -0,0 +1,235 @@
+import type {
+  Company,
+  DataField,
+  Payment,
+  Vehicle,
+  Trip,
+  User,
+} from "./dataFields";
+import { DataLoader } from "./DataLoader";
+import { Role } from "./roles";
+import type { Route } from "./Route";
+import { DataPackage } from "./DataPackage";
+
+export class DataManager {
+  currentRole: Role;
+  currentUser?: User;
+  currentCompany?: Company;
+  currentVehicle?: Vehicle;
+  currentTrip?: Trip;
+
+  companies: Company[];
+  payments: Payment[];
+  trips: Trip[];
+  users: User[];
+  vehicles: Vehicle[];
+  routes: Route[];
+
+  dataLoader: DataLoader;
+  //The currently selected DataPackage
+  currentDataPackage?: DataPackage;
+
+  /**
+   * Construct a new DataManager.
+   */
+  constructor() {
+    //The city is set as the default role
+    this.currentRole = Role.city;
+    this.dataLoader = new DataLoader();
+    this.companies = [];
+    this.payments = [];
+    this.trips = [];
+    this.users = [];
+    this.vehicles = [];
+    this.routes = [];
+
+    this.loadAllData();
+    // this could be problematic here, because loadAllData is async, so
+    // it might not be finished when setAllReference is called.
+    // we might have to use seperate method for loading and setting references outside of the constructor
+    // because the constructor cannot be async (we cannot await this.loadAllData())
+    // Danial: Ja, das ist ein großes Problem, da er ansonsten anföngt die Refernezen zu setzen, ohne das überhaupt die Daten fertig
+    // geladen wurden
+    //this.setAllReferences();
+  }
+
+  /**
+   * This method sets all references that have not been set in the initialization
+   */
+  setAllReferences() {
```

Comment:
Wo sollen wir diese Methode nun ausführen? In load all Data oder einer separaten Methode, die load all Data und set references ausführt?

Reply:
Würde sagen in loadAllData()? Was meinst du/ihr?

Reply:
Ja das hätte ich auch gesagt